### PR TITLE
Unit tests

### PR DESCRIPTION
**Pull request by Hisoka37**
In this update, we wrote unit tests with the `rspec` framework to test all the classes in the application.

### CODE REVIEW

File: spec/person_spec.rb [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+      person = Person.new(18, 'Precious', parent_permission: false)
+
+      expect(person.can_use_services?).to be true
+    end
+
+    it 'should correct a person\'s name' do
+      person = Person.new(18, 'Precious', parent_permission: false)
+
+      expect(person.name).to eql 'Precious'
+    end
```
Comment (BenMukebo) [inline spec/person_spec.rb]: - [ ] Great job so far 👏🏻 👏🏻 However, according to the project requirements, there should be unit tests for all the methods in all the classes, Kindly make sure the `of_age?`, and  `correct_name` class person method has (at least) an 'it' block 👇🏻 check your person.rb file



```ruby 

 def correct_name
    @name
  end

  private

  def of_age?
    @age >= 18
  end
```

**Review by BenMukebo — CHANGES_REQUESTED** [review]
### Hi @Hisoka37, @preciousbetine, @lengushuru,

You did a great job so far 👏🏼 👏🏼
There are some issues that you still need to work on to go to the next project but you are almost there!
<FEEL FREE TO CHANGE THIS MESSAGE TO YOUR PERSONAL WORDS>

### To highlight 💪🏻
- No linter errors. ✔️ 
- Gitflow was correctly used.  ✔️   
- There is one file per class. ✔️ 
 👍  Good job!


## STATUS: CHANGES REQUESTED  ♻️ 
_Please Check the comments under the review and make all required changes_

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._


Cheers and Happy coding!👏👏👏 💻  
Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please ping me @BenMukebo when you comment so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**


------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by BenMukebo — CHANGES_REQUESTED** [review]
### Hi @Hisoka37, @preciousbetine, @lengushuru,

You did a great job so far 👏🏼 👏🏼
There are some issues that you still need to work on to go to the next project but you are almost there!
<FEEL FREE TO CHANGE THIS MESSAGE TO YOUR PERSONAL WORDS>

### To highlight 💪🏻
- No linter errors. ✔️ 
- Gitflow was correctly used.  ✔️   
- There is one file per class. ✔️ 
 👍  Good job!


## STATUS: CHANGES REQUESTED  ♻️ 
_Please Check the comments under the review and make all required changes_

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._


Cheers and Happy coding!👏👏👏 💻  
Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please ping me @BenMukebo when you comment so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**


------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: spec/person_spec.rb (hunk 1 above)
Comment (Hisoka37) [reply spec/person_spec.rb]: @BenMukebo hello ✋🏻 
We have "it" block for `age ` and for `correct-name` , please check `it 'should correct a person\'s name' do` and `it 'should allow a person to use services' do` .
Thank you, we're waiting for your response

Comment (preciousuwhubetine) [reply spec/person_spec.rb]: @BenMukebo, thanks for the review. However, according to best practices, it is advisable to test only public methods hence, the exclusion of `of_age?`.

**Review by codecaiine — APPROVED** [review]
Hi @Hisoka37 @preciousbetine @lengushuru , 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

## Highlights
- You create one file per class ✔️ 
- Test are passed ✔️ 

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me (@codecaiine) in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by nicupop729 — COMMENTED** [review]
Hello @preciousbetine  👋,

_This was supposed to be an additional review as a result of the second opinion request._

As I notice that the project is already approved, this Second Opinion review is redundant.

Keep rocking 🚀!

File: spec/person_spec.rb (hunk 1 above)
Comment (BenMukebo) [reply spec/person_spec.rb]: @Hisoka37, Sorry fo the late reply, I had an emergency  but I'm happy to see that the next code reviewer help you approving this PR, About the the `of_age` it was one of the project requiemrnt to the all the method without specifying (publick and private) but you're also right it is advisable to test only public methods hence, the exclusion of of_age?.

**Comment by preciousuwhubetine** [comment]
@nicupop729 I didn't realize that. My partner must have requested a second opinion.
Thanks for your time!

**Pull request merged**